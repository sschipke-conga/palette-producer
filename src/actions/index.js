

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setAllProjects = allProjects => ({
  type: 'SET_PROJECTS',
  allProjects
});

export const setCurrentPalette = palette => ({
  type: 'SET_CURRENT_PALETTE',
  palette
});

export const setCurrentPaletteInfo = info => ({
  type: 'SET_CURRENT_PALETTE_INFO',
  info
});

export const setCurrentProjectInfo = info => ({
  type: 'SET_CURRENT_PROJECT_INFO',
  info
});

export const setAllPalettes = allPalettes => ({
  type: 'SET_PALETTES',
  allPalettes
});

export const toggleMenu = () => ({
  type: 'TOGGLE_MENU'
})

export const removePalette = (id) => ({
  type: 'REMOVE_PALETTE',
  id: parseInt(id)
});