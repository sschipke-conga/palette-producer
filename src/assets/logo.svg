<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="316.9996" height="316.9996" viewBox="0 0 316.9996 316.9996"><defs><style>.cls-1{fill:#fff;}.cls-1,.cls-2,.cls-3,.cls-4,.cls-5{stroke:#000;stroke-miterlimit:10;stroke-width:10px;}.cls-2{fill:#fbb03b;}.cls-3{fill:#f15a24;}.cls-4{fill:#d4145a;}.cls-5{fill:#3c4892;}</style></defs><title>Artboard 5</title><path class="cls-1" d="M31.62037,143.03632c.33314-55.44444,42.19532-108.237,95.8371-120.86067C185.4136,8.53673,248.78037,35.47176,273.57154,85.89533c31.15582,63.36876,1.52388,129.99623-69.57942,114.45468-13.68126-2.99042-12.8967-5.45845-31.15819-10.9585-23.80811-7.1706-24.20607,3.48171-14.73851,56.5,5,28,7,39-8,49C111.92072,320.34136,31.22091,209.51685,31.62037,143.03632Z"/><circle class="cls-2" cx="75.01795" cy="138.12396" r="22.92254"/><circle class="cls-3" cx="116.7926" cy="76.84227" r="22.92254"/><circle class="cls-4" cx="195.10246" cy="76.8141" r="22.92254"/><circle class="cls-5" cx="230.17288" cy="147.88452" r="22.92254"/></svg>