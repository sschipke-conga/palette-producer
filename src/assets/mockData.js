export const mockPalettes = [
  {
    id: 30,
    project_id: 35,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z",
  },
  {
    id: 33,
    project_id: 35,
    name: "New one now",
    color1: "#f36d6b",
    color2: "#72cc20",
    color3: "#f11a21",
    color4: "#8eb4e8",
    color5: "#b08980",
    created_at: "2020-02-15T00:06:46.617Z",
    updated_at: "2020-02-15T00:06:46.617Z",
  },
  {
    id: 40,
    project_id: 35,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z",
  },
  {
    id: 45,
    project_id: 37,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z"
  },
  {
    id: 43,
    project_id: 37,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z"
  }
]

export const mockProjects = [
  {
    id: 30,
    name: "My First Project",
  },
  {
    id: 33,
    name: "Second Projects",
  },
  {
    id: 40,
    name: "Third Project"
  }
]

export const mockUser = {
  user_id: 34,
  username: 'Gary'
}

export const mockCurrentPalette = [
  {
    name: 'color1',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color2',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color3',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color4',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color5',
    hexCode: '#FFFFFF',
    isLocked: false
  }
]