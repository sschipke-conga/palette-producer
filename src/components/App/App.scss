
@import url('https://fonts.googleapis.com/css?family=Rubik:300,400,500,700,900&display=swap');

body {
  font-family: "Rubik", sans-serif;
}

button {
  cursor: pointer;
}

.App {
  background: rgb(244, 244, 244);
  text-align: center;
}