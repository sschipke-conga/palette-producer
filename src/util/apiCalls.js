export const createNewUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/signup`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 500) {
    throw Error("This username already taken");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};

export const loginUser = async user => {
  let url = `${process.env.REACT_APP_PALETTE_PRODUCER_BACKEND_BASE_URL}/api/v1/login`;
  let options = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  let res = await fetch(url, options);

  if (res.status === 401) {
    throw Error("Username or password is incorrect");
  }

  if (!res.ok) {
    throw Error("Woops! Something went wrong");
  }

  return res.json();
};
