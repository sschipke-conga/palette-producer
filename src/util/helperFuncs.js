export const generateRandomHex = () => {
  let hexCode = "#" + Math.floor(Math.random() * 16777215).toString(16);
  if (hexCode.length < 7) {
    return generateRandomHex();
  }
  return hexCode;
};
