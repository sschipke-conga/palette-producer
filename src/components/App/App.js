import React, { Component } from 'react';
import './App.scss';
import PaletteContainer from '../PaletteContainer/PaletteContainer';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import Modal from '../Modal/Modal';
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { setCurrentPalette, setAllPalettes, setAllProjects } from "../../actions/index";
import { getUserProjects, getProjectPalettes, deletePalette, deleteProject, saveProject, savePalette, updatePalette, updateProject } from '../../util/apiCalls';
import Nav from '../Nav/Nav'
import { Route } from 'react-router-dom';
import allPalettes from '../../reducers/allPalettes';


export class App extends Component {
  constructor() {
    super();
    this.state = {
      userID: null,
      projects: [],
      palettes: {},
      projectName: "",
      paletteName: "",
      currentPalette: {},
      currentProject: {}
    };
  }

  componentDidMount() {
  }

  loadUserProjectsAndPalettes = async userID => {
    const { setAllProjects, setAllPalettes } = this.props
    await this.setState({ projects: [], palettes: {} });
    await this.setState({ userID: userID });
    try {
      let projects = await getUserProjects(userID);
      setAllProjects(projects)
      let palettePromises = [];
      await projects.forEach(project => {
        palettePromises.push(getProjectPalettes(project.id));
      });
      setAllPalettes(await Promise.all(palettePromises));
    } catch ({message}) {
      console.error(message)
      this.setState({projects: [], palettes: {}})
    }
  };

  select = (project, palette) => {
    this.setState({
      currentProject: project,
      projectName: project.name,
      currentPalette: palette,
      paletteName: palette.name
    });
  };

  // removePalette = async e => {
  //   let paletteId = parseInt(e.target.id);
  //   try {
  //     await deletePalette(paletteId);
  //     this.loadUserProjectsAndPalettes(this.state.userID);
  //   } catch ({ message }) { console.error(message) }
  // };

  // removeProject = async e => {
  //   let projectId = parseInt(e.target.id);
  //   try {
  //     await deleteProject(projectId);
  //     this.loadUserProjectsAndPalettes(this.state.userID);
  //   } catch ({ message }) { console.error(message) }
  // };

  // handleChange = e => {
  //   this.setState({ [e.target.name]: e.target.value });
  // };


  render() {
    const {currentPalette, isMenuActive} = this.props
    return (
      <main className="App">
        <Route
          path="/(login|signup)"
          render={() => (
            <div>
              <Nav />
            <Modal loadProjects={this.loadUserProjectsAndPalettes} />
                <PaletteContainer
                userID={this.state.userID}
                currentPalette={currentPalette}
                handleChange={this.handleChange}
                projectName={this.state.projectName}
                paletteName={this.state.paletteName}
                save={this.save}
              />
            </div>
          )}
        />
        <Route
          exact
          path="/"
          render={() => (
            <div>
              <Nav />
              <PaletteContainer
                userID={this.state.userID}
                handleChange={this.handleChange}
                projectName={this.state.projectName}
                paletteName={this.state.paletteName}
                save={this.save}
              />
              {isMenuActive && (
                <ProjectContainer
                  projects={this.state.projects}
                  palettes={this.state.palettes}
                  select={this.select}
                  removePalette={this.removePalette}
                  removeProject={this.removeProject}
                />
              )}
            </div>
          )}
        />
      </main>
    );
  }
}

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setAllPalettes,
      setAllProjects,
      setCurrentPalette,
    },
    dispatch
  );

  export const mapStateToProps = state => ({
    allPalettes: state.allPalettes,
    allProjects: state.allProjects,
    currentPalette: state.currentPalette,
    user: state.user,
    isMenuActive: state.isMenuActive
  });

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {

}