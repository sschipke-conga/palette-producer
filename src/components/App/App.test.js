import React from 'react';
import { shallow } from "enzyme";
import ReactDOM from 'react-dom';
import {App, mapDispatchToProps, mapStateToProps} from './App';
import {mockPalettes, mockProjects, mockUser} from '../../assets/mockData'
import { setAllPalettes, setAllProjects } from '../../actions';
import { getUserProjects, getProjectPalettes } from '../../util/apiCalls';
jest.mock('../../util/apiCalls')

getUserProjects.mockImplementation(() => Promise.resolve(mockProjects));
getProjectPalettes.mockImplementation(() => Promise.resolve(mockPalettes[0]));


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
        setAllPalettes={jest.fn()}
        setAllProjects={jest.fn()}
        isMenuActive={true}
      />)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Alt snapshot', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      setAllPalettes={jest.fn()}
      setAllProjects={jest.fn()}
      isMenuActive={true}
      />)
  });
  it('should match the snapshot when the menu is active ', () => {
    expect(wrapper).toMatchSnapshot();
  });
})

describe('loadUserProjectsAndPalette', () => {
  const mockSetAllPalettes= jest.fn()
  const mockSetAllProjects = jest.fn()
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      setAllPalettes={mockSetAllPalettes}
      setAllProjects={mockSetAllProjects}
      isMenuActive={true}
      user={mockUser}
      />)
  });
  it('should call getUserProjects, getProjectPalettes, setAllProjects, and setAllPalettes', async () => {
    await wrapper.instance().loadUserProjectsAndPalettes()
    expect(getUserProjects).toHaveBeenCalled()
    expect(getProjectPalettes).toHaveBeenCalled()
    expect(mockSetAllPalettes).toHaveBeenCalled()
    expect(mockSetAllProjects).toHaveBeenCalled()
  })
})

describe('App REDUX test', () => {
  it('mapStateToProps gives all the movies in state', () => {
    const mockState = {
      user: mockUser,
      isMenuActive: false
    };
    const expected = {
      user: mockUser,
      isMenuActive: false
    };
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });

  it('calls dispatch with setAllPallettes action when it is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setAllPalettes('SET_PALETTES', mockPalettes);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setAllPalettes('SET_PALETTES', mockPalettes);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('calls dispatch with setAllProjects action when it is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setAllProjects('SET_PROJECTS', mockProjects);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setAllProjects('SET_PROJECTS', mockProjects);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
})