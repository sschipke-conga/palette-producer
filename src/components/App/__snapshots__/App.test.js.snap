// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="App"
>
  <Nav />
  <Route
    path="/(login|signup)"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</main>
`;
